SU2_CFD/src/numerics_direct_turbulent.cpp

```diff
@@ -1095,8 +1095,8 @@ CUpwSca_TurbSST::~CUpwSca_TurbSST(void) {
 
 void CUpwSca_TurbSST::ExtraADPreaccIn() {
   if (incompressible) {
-    AD::SetPreaccIn(V_i, nDim+2);
-    AD::SetPreaccIn(V_j, nDim+2);
+    AD::SetPreaccIn(V_i, nDim+3);
```

Comment:
Have you validated the gradient already for the discrete adjoint using the incompressible solver ?

Reply:
It's looking pretty good so far. Attaching some laminar cylinder checks for 50 FFD vars for both drag and heat flux. I have activated the custom objective and heat objectives for the direct differentiation. I have also added some new objectives, including flow uniformity, secondary flow strength, pressure drop.

![drag](https://user-images.githubusercontent.com/4896083/39510640-382766c2-4d9f-11e8-8023-da9d99ccfdac.png)
![heat](https://user-images.githubusercontent.com/4896083/39510641-38421e90-4d9f-11e8-81c7-012d54664ba1.png)